test_util/aws.py

```diff
@@ -33,6 +47,16 @@ def instances_to_hosts(instances):
     return [Host(i.private_ip_address, i.public_ip_address) for i in instances]
 
 
+def fetch_stack(stack_name, boto_wrapper):
+    stack = boto_wrapper.resource('cloudformation').Stack(stack_name)
+    for resource in stack.resource_summaries.all():
+        if resource.logical_resource_id == 'MasterStack':
+            return DcosZenCfStack(stack_name, boto_wrapper)
+        if resource.logical_resource_id == 'MasterServerGroup':
```

Comment:
I do not follow your question. These terms are dictated by our own AWS templates. In 'simple deploys' we have a single stack with everything jammed in. This means there must be a master auto scaling group (no guarantee about agents in any scenario). If this is a zen stack, then the autoscaling groups are segregated in their own private stacks. If the arbitrary logical ID's that we put in our templates change, then this will break.